// @mantine/nprogress/cjs/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var NavigationProgress = require('./NavigationProgress.js');
var nprogress_store = require('./nprogress.store.js');



exports.NavigationProgress = NavigationProgress.NavigationProgress;
exports.cleanupNavigationProgress = nprogress_store.cleanupNavigationProgress;
exports.cleanupNavigationProgressAction = nprogress_store.cleanupNavigationProgressAction;
exports.completeNavigationProgress = nprogress_store.completeNavigationProgress;
exports.completeNavigationProgressAction = nprogress_store.completeNavigationProgressAction;
exports.createNprogress = nprogress_store.createNprogress;
exports.createNprogressStore = nprogress_store.createNprogressStore;
exports.decrementNavigationProgress = nprogress_store.decrementNavigationProgress;
exports.decrementNavigationProgressAction = nprogress_store.decrementNavigationProgressAction;
exports.incrementNavigationProgress = nprogress_store.incrementNavigationProgress;
exports.incrementNavigationProgressAction = nprogress_store.incrementNavigationProgressAction;
exports.nprogress = nprogress_store.nprogress;
exports.nprogressStore = nprogress_store.nprogressStore;
exports.resetNavigationProgress = nprogress_store.resetNavigationProgress;
exports.resetNavigationProgressAction = nprogress_store.resetNavigationProgressAction;
exports.setNavigationProgress = nprogress_store.setNavigationProgress;
exports.setNavigationProgressAction = nprogress_store.setNavigationProgressAction;
exports.startNavigationProgress = nprogress_store.startNavigationProgress;
exports.startNavigationProgressAction = nprogress_store.startNavigationProgressAction;
exports.stopNavigationProgress = nprogress_store.stopNavigationProgress;
exports.stopNavigationProgressAction = nprogress_store.stopNavigationProgressAction;
exports.updateNavigationProgressStateAction = nprogress_store.updateNavigationProgressStateAction;
exports.useNprogress = nprogress_store.useNprogress;
